x-pack/plugins/security/public/nav_control/nav_control_service.tsx

```diff
@@ -9,11 +9,12 @@ import { CoreStart } from 'src/core/public';
 import ReactDOM from 'react-dom';
 import React from 'react';
 import { SecurityLicense } from '../../common/licensing';
-import { AuthenticatedUser } from '../../common/model';
 import { SecurityNavControl } from './nav_control_component';
+import { AuthenticationServiceSetup } from '../authentication';
 
 interface SetupDeps {
   securityLicense: SecurityLicense;
+  getCurrentUser: AuthenticationServiceSetup['getCurrentUser'];
 }
```

Comment:
I'm just curious, is there any particular reason you decided to use such a specific contract? 

My 2 cents is that it would have sufficed to just use `AuthenticationServiceSetup` as a parameter -- less verbose that way, and less TypeScript jargon / easier to understand for new developers.
I noticed that `SecurityLicense` is also more generic (e.g., we don't specify `SecurityLicense['feature$']` here, even though we could).

Reply:
I'm asking myself the same question in my next PR that is based on this one - and I'm totally agree that it looks weird and unnecessary. Can't recall why I did this, must be `I'll think about it later, but never get back to it....` kind of thing :see_no_evil: I'll fix it, thanks!